File: Dataset/Leetcode/train/12/77.java
 class Solution {
    public String XXX(int num) {
        int[] nums = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        String[] romans = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};

        StringBuilder string = new StringBuilder();
        int index = 0;
        while(index < 13) {
            while(num >= nums[index]){
                string.append(romans[index]);
                num -= nums[index]; 
            }
            index++;
        }
        return string.toString();
    }
}

File: Dataset/Leetcode/train/3/77.py
 class Solution:
    def XXX(self, s: str) -> int:
        list1 = []
        len_ = 0
        for i in s:
            if i in list1:
                if len_ < len(list1):
                    len_ = len(list1)
                list1 = list1[list1.index(i) + 1:]
            list1.append(i)
            if len_ < len(list1):
                len_ = len(list1)
        return len_


